src/controllers/auth.controller.js

```diff
@@ -0,0 +1,325 @@
+import {
+  usersService,
+  emailService,
+  jwtService,
+  tokenService,
+} from './../services/index.js';
+import { userSchema } from './../schemas/index.js';
+import { ApiError } from '../exeptions/api.error.js';
+import { dynamicSchema } from './../utils/dynamicSchema.js';
+import bcrypt from 'bcrypt';
+
+const register = async (req, res) => {
+  const userInformation = req.body;
+
+  const { error, value } = userSchema.validate(userInformation, {
+    allowUnknown: true,
+  });
+
+  const hashedPassword = await bcrypt.hash(value.password, 10);
```

Comment:
Ensure that `value.password` is defined before attempting to hash it. If `value.password` is undefined, this will throw an error.